receiver/kafkareceiver/config.go

```diff
@@ -96,5 +99,8 @@ var _ component.Config = (*Config)(nil)
 
 // Validate checks the receiver configuration is valid
 func (cfg *Config) Validate() error {
+	if len(cfg.Topic) > 0 && len(cfg.TopicRegex) > 0 {
```

Comment:
I wonder if it is more appropriate to use the logic of priority

Reply:
This might be right I changed that. I removed this validation rule. Remaining code works in such a way that when `TopicRegex` is configured we start looking for new topics which match the regex and if they are detected we overwrite list of topics that we subscribe: [link](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36909/files#diff-8fa2a92bedeb4eca30c174611f6204e98b4d80570d7b0c37a3c8b2fdf4e75ef7R249-R252)